src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.LeakDetection/CheckForPoison.cs

```diff
@@ -288,7 +290,14 @@ private static PoisonedFileEntry CheckSingleFile(IEnumerable<CatalogPackageEntry
                 AssemblyName asm = AssemblyName.GetAssemblyName(fileToCheck);
                 if (IsAssemblyPoisoned(fileToCheck))
                 {
-                    poisonEntry.Type |= PoisonType.AssemblyAttribute;
+                    if (fileToCheck.EndsWith(".dll") && HasAttributeOfType(fileToCheck, "System.Runtime.CompilerServices.ReferenceAssemblyAttribute"))
```

Comment:
It doesn't seem correct to check for ReferenceAssemblyAttribute here.  We are trying to catch SBRP leaks here - the PoisonType.ReferenceAssemblyAttribute really should be tied to SBRP.  It is possible for non-SBRP ref assembly leaks.  These should not be classified the same IMO.  For example the $(NetCurrent) targeting pack is in the n-1 artifacts and they could be leaked.  We should treat that as a normal n-1 assembly leak IMO.  The primary reason for thinking this is the resolution is likely different between these two situations.

This core logic is what we can use but IMO we should be checking for the SBRP marker attribute that should be added back first.  That will also eliminate the need to "poison" the SBRPs because that attribute in itself can be used as the poison marker.

Reply:
Updated in https://github.com/dotnet/installer/commit/7e4372eb95f9bbc49cbd9db34159c9332c68f52c